tests/test_web.py

```diff
@@ -663,6 +663,24 @@ def test_cached_response_unchanged(web_response_mock, oauth_client, mock_time):
     assert result.status_unchanged
 
 
+@responses.activate
+def test_non_idempotent_request_methods(
+    web_response_mock, oauth_client, mock_time
+):
+    responses.add(responses.POST, "https://api.spotify.com/v1/foo", json={})
+    responses.add(responses.PUT, "https://api.spotify.com/v1/foo", json={})
+    responses.add(responses.DELETE, "https://api.spotify.com/v1/foo", json={})
+    oauth_client._expires = 2000
+    mock_time.return_value = 1
+
+    result = oauth_client.post("foo")
+    result = oauth_client.put("foo")
+    result = oauth_client.delete("foo")
+
+    assert len(responses.calls) == 3
+    assert not result.status_unchanged
```

Comment:
This only tests the final result you got from `delete(). And what is this trying to test? It also passes if the last one is changed to `result = oauth_client.get("foo")`.

Reply:
oh wow, that test was a complete brainfart, sorry for that. it is supposed to verify that non-idempotent http methods don't use the cache. rewritten and works now (also counterchecked that `.get` fails at this position.